commons-test/src/main/java/org/infinispan/commons/test/PolarionTestNameListenerAdapter.java

```diff
@@ -0,0 +1,127 @@
+package org.infinispan.commons.test;
+
+import java.lang.reflect.Field;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import org.testng.ITestContext;
+import org.testng.ITestResult;
+import org.testng.TestListenerAdapter;
+import org.testng.annotations.Test;
+
+/**
+ * Friendly test name
+ *
+ * @author <a href='mailto:afield[at]redhat[dot]com'>Alan Field</a>
+ * @author <a href='mailto:dlovison[at]redhat[dot]com'>Diego Lovison</a>
+ */
+public class PolarionTestNameListenerAdapter extends TestListenerAdapter {
```

Comment:
Do you really need to extend `TestListenerAdapter`? I don't think you need to keep track of all the tests that have run any more.

And longer javadoc to explain what you're doing and why might help in the future when we upgrade TestNG and this stops working.

Reply:
@diegolovison did you forget to push the change to implement `IResultListener2` instead of extending `TestListenerAdapter`? You could also implement only `ITestListener`, it has fewer methods and I don't think you need anything from `IResultListener2`. Also the class name shouldn't end in `Adapter`, because it's just a listener, it's not adapting another listener.

Reply:
I implemented `ITestListener`. `Adapter` from the class name was removed.